skyportal/facility_apis/ztf.py

```diff
@@ -206,6 +363,123 @@ def delete(request):
+            Instrument.query_records_accessible_by(req.requester)
+            .join(Allocation)
+            .join(FollowupRequest)
+            .filter(FollowupRequest.id == request.id)
+            .first()
+        )
+
+        altdata = request.allocation.altdata
+
+        if not altdata:
+            raise ValueError('Missing allocation information.')
+
+        keys = ['ra', 'dec', 'jdstart', 'jdend']
+
+        content = req.transactions[-1].response["content"]
+        df_request = pd.read_html(content)[0]
+        if df_request.shape[0] == 0:
+            raise ValueError('Missing response from forced photometry service.')
+        df_request.columns = df_request.columns.str.lower()
+        if not set(keys).issubset(df_request.columns):
+            raise ValueError("RA, Dec, jdstart, and jdend required in response.")
+        df_request = df_request.iloc[0]
+        df_request = df_request.replace({np.nan: None})
+
+        requestgroup = {
+            "email": altdata["ipac_email"],
+            "userpass": altdata["ipac_userpass"],
+            "option": "All recent jobs",
+            "action": "Query Database",
+        }
+        params = urllib.parse.urlencode(requestgroup)
+        url = f"{ZTF_FORCED_URL}/cgi-bin/getForcedPhotometryRequests.cgi?{params}"
+
+        r = requests.get(
+            url,
+            auth=HTTPBasicAuth(
+                altdata['ipac_http_user'], altdata['ipac_http_password']
+            ),
+        )
+        r.raise_for_status()
+
+        if r.status_code == 200:
+            df_result = pd.read_html(r.text)[0]
+            df_result.rename(
+                inplace=True, columns={'startJD': 'jdstart', 'endJD': 'jdend'}
+            )
+            df_result = df_result.replace({np.nan: None})
+            if not set(keys).issubset(df_result.columns):
+                raise ValueError("RA, Dec, jdstart, and jdend required in response.")
+            index_match = None
+            for index, row in df_result.iterrows():
+                if not all([np.isclose(row[key], df_request[key]) for key in keys]):
+                    continue
+                index_match = index
+            if index_match is None:
+                raise ValueError(
+                    'No matching response from forced photometry service. Please try again later.'
+                )
+            else:
+                row = df_result.loc[index_match]
+                if row['lightcurve'] is None:
+                    raise ValueError(
+                        'Light curve not yet available. Please try again later.'
+                    )
+                else:
+                    lightcurve = row['lightcurve']
+                    dataurl = f"{ZTF_FORCED_URL}/{lightcurve}"
+                    IOLoop.current().run_in_executor(
+                        None,
+                        lambda: commit_photometry(
+                            dataurl, altdata, df_request, req.id, instrument.id
+                        ),
```

Comment:
So this GET request is going to be posting things to the DB? I'm not sure this is kosher. @stefanv what do you think about this?

Reply:
The naming is a bit funny because we are getting the data after posting a request to their server.